src/main/java/uk/co/bconline/ndelius/transformer/UserTransformer.java

```diff
@@ -146,6 +181,14 @@ public Optional<User> map(UserEntry user)
 
 	public ExportResult combine(UserExportEntity db, UserEntry ldap) {
 		if (ldap == null) return null;
+		DateTimeFormatter format = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss");
+		Set<RoleEntry> userRoles;
+		try {
+			userRoles = userRoleService.getUserRoles(ofNullable(ldap.getUsername()).orElseGet(db::getUsername));
+		} catch (Exception e)
+		{
+			userRoles = new HashSet<>();
+		}
```

Comment:
Do we need this try-catch - what exception are we expecting here?  I don't think we'll ever need to pass the DB username either, as it's an LDAP lookup - would this be simpler:
```suggestion
		val userRoles = ofNullable(ldap.getUsername()).map(userRoleService::getUserRoles).orElse(emptySet())
```

Also, I think fetching the roles for each user during the export might kill the performance.  But I guess we won't know till we try it in pre-prod...